src/main/java/org/foobaz42/compareverifier/ComparableVerifier.java

```diff
@@ -202,32 +204,60 @@ public void verify() {
 
         // verify that the returned instances return false when checked for equality
         // with null
-        verifyEqualsToNullReturnsFalse(lessInstances);
+        verifyEqualsToNullReturnsFalse(lesserInstances);
         verifyEqualsToNullReturnsFalse(equalInstances);
         verifyEqualsToNullReturnsFalse(greaterInstances);
 
         // verify that the returned instances throw an exception when compared
         // to null
-        verifyExceptionOnCompareToNull(lessInstances);
+        verifyExceptionOnCompareToNull(lesserInstances);
         verifyExceptionOnCompareToNull(equalInstances);
         verifyExceptionOnCompareToNull(greaterInstances);
 
         // verify that sgn(a.compareTo(b)) == -sgn(b.compareTo(a))
         verifyReverse(equalInstances, equalInstances);
-        verifyReverse(equalInstances, lessInstances);
+        verifyReverse(equalInstances, lesserInstances);
         verifyReverse(equalInstances, greaterInstances);
-        verifyReverse(lessInstances, greaterInstances);
+        verifyReverse(lesserInstances, greaterInstances);
+
+        // verify that sgn(a.compareTo(b)) > 0 && sgn(b.compareTo(c)) > 0 => sgn(a.compareTo(c)) > 0
+        verifyTransitivity(lesserInstances, equalInstances, greaterInstances);
 
-        // TODO: testing transitivity:
-        // TODO: test sgn(a.compareTo(b)) > 0 && sgn(b.compareTo(c)) > 0 => sgn(a.compareTo(c)) > 0
         // TODO: test sgn(a.compareTo(c)) == sgn(b.compareTo(c)) => sgn(a.compareTo(b)) == 0
     }
 
+
+    // sgn(a.compareTo(b)) > 0 && sgn(b.compareTo(c)) > 0 => sgn(a.compareTo(c)) > 0
+    private void verifyTransitivity(final List<A> lesser,
+                                    final List<A> equal,
+                                    final List<A> greater) {
+        for (final A la : lesser) {
+            for (final A ea : equal) {
+                for (final A ga : greater) {
+                    final int equal_lesser = Math2.sgn(ea.compareTo(la));
+                    final int greater_equal = Math2.sgn(ga.compareTo(ea));
+                    final int greater_lesser = Math2.sgn(ga.compareTo(la));
+
+                    final boolean isTransitive =
+                            equal_lesser > 0
+                                    && greater_equal > 0
+                                    && greater_lesser > 0;
+
+                    if (!isTransitive) {
+                        throw new AssertionError(
+                                format("Instances %s, %s, %s, are not transitive!", la, ea, ga)
```

Comment:
I am wondering about the format of the exception that is thrown here. I've added the information about instances. Should we specify in the docs that for debugging purposes instances should have a `toString` implemented?

Reply:
I think we should specify everything which we can, it help us to reduce the number of issues because documentation will describe everything.